src/server/server_family.cc

```diff
@@ -3605,6 +3605,10 @@ void ServerFamily::ReplTakeOver(CmdArgList args, const CommandContext& cmd_cntx)
   if (IsClusterEnabled()) {
     service().cluster_family().ReconcileMasterReplicaTakeoverSlots(false);
   }
+
+  // Start journal to allow partial sync from same source master
```

Comment:
how it worked before?! We tested partial sync, so how it worked without journal starting?